README.md

```diff
@@ -5,58 +5,83 @@
 
 ## Table of Contents
 * [About](#about)
+* [Share your project!](#share-your-project)
 * [Local Setup](#local-setup)
   * [Installing Required Tools](#installing-required-tools)
+    * [1. Poetry](#1-poetry)
+    * [2. Node.js, npm and pnpm](#2-nodejsm-npm-and-pnpm)
+    * [3. Docker](#3-docker)
+    * [4. Docker Compose](#4-docker-compose)
   * [Setting Up Environment Variables](#setting-up-environment-variables)
   * [If you are not using Docker](#if-you-are-not-using-docker)
     * [Backend](#backend)
     * [Frontend](#frontend)
   * [If you are using Docker](#if-you-are-using-docker)
 * [Pre-Commit Setup](#pre-commit-setup)
+  * [Installing and Activating Pre-Commit Hooks](#installing-and-activating-pre-commit-hooks)
+  * [Running Pre-Commit Checks](#running-pre-commit-checks)
+  * [Updating Pre-Commit Hooks](#updating-pre-commit-hooks)
 * [Running the Application](#running-the-application)
 * [Watchers](#watchers)
```

Comment:
I'm not sure if watchers is the best name for what we have. Perhaps call it hot-reload or something similar.